go/vt/vttablet/tabletmanager/tm_init.go

```diff
@@ -486,6 +487,33 @@ func (tm *TabletManager) createKeyspaceShard(ctx context.Context) (*topo.ShardIn
 	}); err != nil {
 		return nil, vterrors.Wrap(err, "createKeyspaceShard: cannot GetOrCreateShard shard")
 	}
+
+	// Ensure that this tablet comes up with the sidecar database
+	// name that is set for the keyspace.
+	ks, err := tm.TopoServer.GetKeyspace(ctx, tablet.Keyspace)
+	if err != nil {
+		return nil, vterrors.Wrap(err, "createKeyspaceShard: cannot GetOrCreateShard shard")
+	}
+	// If the keyspace exists but this is the first tablet added, then
+	// update the keyspace record to the default.
+	if ks.SidecarDbName == "" {
+		ks.SidecarDbName = sidecardb.DefaultName
+		ctx, unlock, lockErr := tm.TopoServer.LockKeyspace(ctx, tablet.Keyspace, "Setting sidecar database name")
+		if lockErr != nil {
+			return nil, vterrors.Wrap(lockErr, "createKeyspaceShard: cannot GetOrCreateShard shard")
+		}
+		err = tm.TopoServer.UpdateKeyspace(ctx, ks)
+		unlock(&lockErr)
+		if err != nil {
+			return nil, vterrors.Wrap(err, "createKeyspaceShard: cannot GetOrCreateShard shard")
+		}
+		if lockErr != nil {
+			return nil, vterrors.Wrap(lockErr, "createKeyspaceShard: cannot GetOrCreateShard shard")
+		}
```

Comment:
if there is lock error we will not bring up the tablet.
lock error can be multiple tablets trying to update the same thing here and lock timeout.
I think with lock timeout we should add a retry.


what I am trying to say here is. Some thoughts should be when multiple tablets come at the same time, will this cause any issue or not

Reply:
If you have a non-serving keyspace, a keyspace with no tablets, and then you simultaneously bring up 10 tablets let's say... it's possible. I'll work on something for this.